app/Services/Rogue.php

```diff
@@ -110,9 +110,9 @@ public function getActionIdFromCallPowerCampaignId($callpowerCampaignId)
             'filter' => ['callpower_campaign_id' => $callpowerCampaignId],
         ]);
 
-        if (! $action['data'][0]['id']) {
+        if (empty($action['data'])) {
             info('action id not found for CallPower campaign id: ' . $callpowerCampaignId);
-            throw new Exception(500, 'Unable to get action data for CallPower campaign id : ' . $callpowerCampaignId);
+            abort(500, 'Unable to get action data for CallPower campaign id: ' . $callpowerCampaignId);
```

Comment:
I think (another) reason we wouldn't have been seeing these errors in the logs was that we were using the wrong signature for the [`Exception`](https://www.php.net/manual/en/class.exception.php) constructor (the first parameter is the message, not the error code).

And this is sorta nitpicky, but I wonder if throwing an old-school Exception is preferable to the `abort` helper here, since `abort` throws a `HttpException` with a status code & this isn't actually running within a web request...